great_expectations/experimental/datasources/experimental_base_model.py

```diff
@@ -55,12 +84,68 @@ def yaml(
 
         # pydantic json encoder has support for many more types
         # TODO: can we dump json string directly to yaml.dump?
-        intermediate_json = json.loads(self.json())
+        intermediate_json = self._json_dict(
+            include=include,
+            exclude=exclude,
+            by_alias=by_alias,
+            skip_defaults=skip_defaults,
+            exclude_unset=exclude_unset,
+            exclude_defaults=exclude_defaults,
+            exclude_none=exclude_none,
+            encoder=encoder,
+            models_as_dict=models_as_dict,
+        )
         yaml.dump(intermediate_json, stream=stream_or_path, **yaml_kwargs)
 
         if isinstance(stream_or_path, pathlib.Path):
             return stream_or_path
         return stream_or_path.getvalue()
 
+    def _json_dict(
+        self,
+        *,
+        include: Union[AbstractSetIntStr, MappingIntStrAny, None] = None,
+        exclude: Union[AbstractSetIntStr, MappingIntStrAny, None] = None,
+        by_alias: bool = False,
+        skip_defaults: Union[bool, None] = None,
+        exclude_unset: bool = False,
+        exclude_defaults: bool = False,
+        exclude_none: bool = False,
+        encoder: Union[Callable[[Any], Any], None] = None,
+        models_as_dict: bool = True,
+        **dumps_kwargs,
+    ) -> dict:
+        """
+        JSON compatible dictionary. All complex types removed.
+        Prefer `.dict()` or `.json()`
+        """
+        return json.loads(
+            self.json(
+                include=include,
+                exclude=exclude,
+                by_alias=by_alias,
+                skip_defaults=skip_defaults,
+                exclude_unset=exclude_unset,
+                exclude_defaults=exclude_defaults,
+                exclude_none=exclude_none,
+                encoder=encoder,
+                models_as_dict=models_as_dict,
+                **dumps_kwargs,
+            )
+        )
+
     def __str__(self):
         return self.yaml()
+
+    def __copy__(self):
+        return self.copy(deep=False)
+
+    def __deepcopy__(self, memo):
+        LOGGER.debug(f"{type(self).__name__} __deep_copy__")
+        warnings.warn(
+            f"{type(self).__name__}.__deepcopy__ returns a shallow copy",
+            ZEPDeepCopyWarning,
+        )
+        copy = self.copy(deep=False)
+        memo[id(copy)] = copy
```

Comment:
I'm not sure this does anything.
If we decide to merge this PR I will test and remove it if possible.